src/tpm.rs

```diff
@@ -613,10 +605,10 @@ fn hash_alg_to_message_digest(
 fn check_if_pcr_data_and_attestation_match(
     hash_algo: HashingAlgorithm,
     pcr_data: &PcrData,
-    attestation: &TPM2B_ATTEST,
+    attestation: Attest,
 ) -> Result<bool> {
-    let pcr_data = TPML_DIGEST::try_from(pcr_data.clone())?;
-    let attestation = unmarshal_tpms_attest(&attestation.attestationData)?;
+    let pcr_data = Vec::<TPML_DIGEST>::try_from(pcr_data.clone())?;
+    let attestation = TPMS_ATTEST::from(attestation);
```

Comment:
I suspect this function could be refactored to use the public methods on the `Attest` structure, something like:
```rust
fn check_if_pcr_data_and_attestation_match(
    hash_algo: HashingAlgorithm,
    pcr_data: &PcrData,
    attestation: &Attest,
) -> Result<bool> {
    let pcr_data = Vec::<TPML_DIGEST>::try_from(pcr_data.clone())?;

    let quote = match attestation.attested() {
        AttestInfo::Quote { info } => info,
        _ => {
            return Err(KeylimeError::Other(format!(
                "Expected attestation type TPM2_ST_ATTEST_QUOTE, got {:?}",
                attestation.attestation_type()
            )));
        }
    };

    let attested_pcr = quote.pcr_digest().value();
```
Then you could omit a `clone` on Attest, marshalling to `TPMS_ATTEST`, and `unsafe` block around `attestation.attested.quote`.

Reply:
Thanks @ueno , this was a good suggestion! I've made the update.